pkg/extension/sumologicextension/extension.go

```diff
@@ -223,17 +223,40 @@ func (se *SumologicExtension) Shutdown(ctx context.Context) error {
 func (se *SumologicExtension) validateCredentials(
 	ctx context.Context,
 	colCreds credentials.CollectorCredentials,
-) error {
+) (recoverable bool, err error) {
 	se.logger.Info("Validating collector credentials...",
 		zap.String(collectorCredentialIdField, colCreds.Credentials.CollectorCredentialId),
 		zap.String(collectorIdField, colCreds.Credentials.CollectorId),
 	)
 
 	if err := se.injectCredentials(colCreds); err != nil {
-		return err
+		return false, err
 	}
 
-	return se.sendHeartbeatWithHTTPClient(ctx, se.httpClient)
+	se.backOff.Reset()
+	for {
+		recoverable, err = se.sendHeartbeatWithHTTPClient(ctx, se.httpClient)
```

Comment:
Not sure if recoverable is a good name, as this value means that that we get error which is API unrelated (network, conenctivity, API unavailability)

Reply:
On the other hand if the error if this is final state, we want to stop the collector

Reply:
Can't we inspect the error itself instead of adding a boolean return value? Looks like the only error on which we want to fail is `errUnauthorizedHeartbeat`.

Reply:
The way I imagine this logic would be:
- `sendHeartbeatWithHTTPClient` always retries on temporary errors
- `validateCredentials` inspects the returned error and either fails the check if it's a 401, or causes the extension to fail completely

Reply:
Not sure about that statement. what about other 4xx errors?

Reply:
I was considering using extended error which would contain that boolean information inside, but this is the same solution basically